### PR TITLE
79984 improved HttpClient setup for integration tests

### PR DESCRIPTION
**Pull request by janid1967**
See commit msgs

### CODE REVIEW

File: src/tests/Equinor.Procosys.Preservation.WebApi.IntegrationTests/TestFactory.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
 namespace Equinor.Procosys.Preservation.WebApi.IntegrationTests
 {
-    public class TestFactory : WebApplicationFactory<Startup>
+    public sealed class TestFactory : WebApplicationFactory<Startup>
     {
         private readonly string _libraryAdminOid = "00000000-0000-0000-0000-000000000001";
```
Comment (hbjorgo) [inline src/tests/Equinor.Procosys.Preservation.WebApi.IntegrationTests/TestFactory.cs:33]: Consider moving the OIDs together with the user types.

File: src/tests/Equinor.Procosys.Preservation.WebApi.IntegrationTests/TestFactory.cs [hunk 2]
```diff
@@ -63,7 +58,28 @@ public class TestFactory : WebApplicationFactory<Startup>
...
+                {  
+                    lock (s_padlock)  
+                    {  
+                        if (s_instance == null)  
+                        {  
+                            s_instance = new TestFactory();  
+                        }  
+                    }  
+                }  
+                return s_instance;            }
```
Comment (hbjorgo) [inline src/tests/Equinor.Procosys.Preservation.WebApi.IntegrationTests/TestFactory.cs:79]: Bracket formatting.

File: src/tests/Equinor.Procosys.Preservation.WebApi.IntegrationTests/TestFactory.cs [hunk 3]
```diff
@@ -112,6 +112,9 @@ public new void Dispose()
             base.Dispose();
         }
 
+        public HttpClient GetHttpClient(UserType userType, string plant)
```
Comment (hbjorgo) [inline src/tests/Equinor.Procosys.Preservation.WebApi.IntegrationTests/TestFactory.cs:115]: Any need for this method? Why not make the target method public?

File: src/tests/Equinor.Procosys.Preservation.WebApi.IntegrationTests/TestFactory.cs (hunk 1 above)
Comment (hbjorgo) [reply src/tests/Equinor.Procosys.Preservation.WebApi.IntegrationTests/TestFactory.cs:33]: Perhaps to where the names etc are set?

Comment (janid1967) [reply src/tests/Equinor.Procosys.Preservation.WebApi.IntegrationTests/TestFactory.cs:33]: The main reason to put them here, at same place, is to be "sure" to make them unique

Comment (janid1967) [reply src/tests/Equinor.Procosys.Preservation.WebApi.IntegrationTests/TestFactory.cs:33]: won't fix

File: src/tests/Equinor.Procosys.Preservation.WebApi.IntegrationTests/TestFactory.cs (hunk 3 above)
Comment (janid1967) [reply src/tests/Equinor.Procosys.Preservation.WebApi.IntegrationTests/TestFactory.cs:117]: You're right. This method was static a while, but when refactoring it to be non-static it became superfluous

**Review by hbjorgo — APPROVED** [review]

**Pull request merged**